### PR TITLE
Refactoring code and new features in 0.6.0-beta.6

### PR DESCRIPTION
**Pull request by sarveshnagpal**
- Refactoring global variable from clarity.* -> clarity(*) to queue up events on the client
- Enable unmasking through configuration
- Suppress selectors using config in addition to existing support for HTML attribute
- Simplify configuration and move internal settings to const enum
- Add event summary
- Add checks for limits and fail safe settings
- Log when we hit a limit and that causes Clarity to stop
- Add hash to each element during visualization
- Change metadata function as a callback

### CODE REVIEW

File: packages/clarity-js/types/core.d.ts [hunk 1]
```diff
@@ -16,17 +16,25 @@ export const enum Time {
     Hour = 60 * 60 * 1000
 }
 
+export const enum Setting {
+    LongTask = 30, // 30ms
+}
```
Comment (swaathee) [inline packages/clarity-js/types/core.d.ts:21]: why not combine with the other setting enum?

File: packages/clarity-js/src/layout/dom.ts [hunk 2]
```diff
@@ -42,18 +43,27 @@ function reset(): void {
...
+        for (const key of Object.keys(config.regions)) {
             let element = root.querySelector(config.regions[key]);
             if (element) { regionMap.set(element, key); }
         }
+
+        // Extract nodes with explicit masked configuration
+        for (const entry of config.suppress) {
+            let element = root.querySelector(entry);
+            if (element) { maskedMap.set(element, true); }
+        }
```
Comment (swaathee) [inline packages/clarity-js/src/layout/dom.ts:66]: what happens when mask is true and suppress is also set in config? : do we need to think about unmasking some elements only based on config?

File: packages/clarity-js/src/data/upload.ts [hunk 3]
```diff
@@ -43,10 +42,11 @@ export function queue(tokens: Token[], transmit: boolean = true): void {
         switch (type) {
             case Event.Discover:
             case Event.Mutation:
-                // At the moment, we limit playback to grow until MAX_PLAYBACK_BYTES. Anytime we grow past this size, we start dropping events.
+                // At the moment, we limit playback to grow until MAX_PLAYBACK_BYTES.
+                // Anytime we grow past this size, we start dropping events.
                 // This is not ideal, and more of a fail safe mechanism.
```
Comment (swaathee) [inline packages/clarity-js/src/data/upload.ts:47]: nit : change comment to reflect recent changes

File: packages/clarity-js/src/data/limit.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        if (reason !== data.check) {
+            trigger(reason);
+        }
+    }
+}
+
+export function trigger(reason: Check): void {
+    data.check = reason;
+    clarity.stop();
+}
```
Comment (swaathee) [inline packages/clarity-js/src/data/limit.ts:28]: how do we ensure compute is called before this?

Comment (sarveshnagpal) [reply packages/clarity-js/src/data/limit.ts:28]: Discussed offline. Through module ordering in data/index.ts file.

File: packages/clarity-js/src/data/upload.ts (hunk 3 above)
Comment (sarveshnagpal) [reply packages/clarity-js/src/data/upload.ts:47]: Thanks, fixed.

**Pull request merged**